src/Dialect/ONNX/Transforms/Decompose.cpp

```diff
@@ -668,37 +669,548 @@ bool shouldDecomposeConvTransposeOp(Value convTransposeResult) {
-bool shouldDecomposeConvTransposeOpTo4Conv(Value convTransposeResult,
+bool ShouldDecomposeConvTransposeOpToPhasedConvs(Value convTransposeResult,
     ArrayAttr kernelShapeAttr, ArrayAttr padsShapeAttr,
-    ArrayAttr stridesShapeAttr) {
-  if (!onnx_mlir::enableConvTranposeDecomposeTo4Conv) {
+    ArrayAttr stridesShapeAttr, ArrayAttr outputShapeAttr) {
+  if (!onnx_mlir::enableConvTranposeDecomposeToPhasedConv) {
     // Disable the ONNXConvTransposeOp to Conv decomposition patterns.
     return false;
   }
-  auto isSymmetricEvenAttribute = [](ArrayAttr arrayAttr,
-                                      bool checkForNonZero) -> bool {
+
+  ONNXConvTransposeOp op =
+      mlir::cast<ONNXConvTransposeOp>(convTransposeResult.getDefiningOp());
+  bool areSpatialDimsStatic = hasShapeAndRank(convTransposeResult) &&
+                              hasStaticSpatialDims(op.getX()) &&
+                              hasStaticSpatialDims(op.getW());
+  if (!areSpatialDimsStatic)
+    return false;
+  auto getVectorFromArrayAttr =
+      [](ArrayAttr arrayAttr) -> SmallVector<int64_t, 4> {
     assert(mlir::dyn_cast<IntegerAttr>(arrayAttr.getValue()[0]) &&
            "Attribute must be integer");
     int nElements = arrayAttr.getValue().size();
     SmallVector<int64_t, 4> elements(nElements, 0);
     for (int i = 0; i < nElements; ++i) {
       elements[i] = mlir::cast<IntegerAttr>(arrayAttr.getValue()[i]).getInt();
     }
-    bool isSymmetricEven = std::all_of(elements.begin(), elements.end(),
-        [&elements, &checkForNonZero](int64_t i) {
-          return (i == elements[0]) && (i % 2 == 0) &&
-                 (checkForNonZero ? elements[0] != 0 : true);
-        });
-    return isSymmetricEven;
+    return elements;
   };
+  // kernel shape is required for decomposition, Do not support the case where
+  // kernel shape can be inferred. Do not support the case where pad values are
+  // to be inferred automatically from outputShape.
+  if (!kernelShapeAttr || outputShapeAttr) {
+    return false;
+  }
 
-  ONNXConvTransposeOp op =
-      mlir::cast<ONNXConvTransposeOp>(convTransposeResult.getDefiningOp());
-  return hasShapeAndRank(convTransposeResult) &&
-         hasStaticSpatialDims(op.getX()) && hasStaticSpatialDims(op.getW()) &&
-         isSymmetricEvenAttribute(kernelShapeAttr, false) &&
-         isSymmetricEvenAttribute(padsShapeAttr, true) &&
-         isSymmetricEvenAttribute(stridesShapeAttr, false);
+  auto kernelShape = getVectorFromArrayAttr(kernelShapeAttr);
+  auto padsShape = (padsShapeAttr) ? getVectorFromArrayAttr(padsShapeAttr)
+                                   : SmallVector<int64_t, 4>({0, 0, 0, 0});
+  auto stridesShape = (stridesShapeAttr)
+                          ? getVectorFromArrayAttr(stridesShapeAttr)
+                          : SmallVector<int64_t, 4>({1, 1});
+  ONNXConvTransposeOpShapeHelper convTransposeShapeHelper(
+      op.getOperation(), {});
+  convTransposeShapeHelper.computeShapeAndAssertOnFailure();
```

Comment:
```suggestion
  convTransposeShapeHelper.computeShapeAndAssertOnFailure();
```
We should handle this by returning `false`, not by asserting

Reply:
Removed the usage of shapeHelper.